src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,24 +1,48 @@
 package pureconfig
 
-trait ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String
-}
-
-class IdentityConfigFieldMapping[T] extends ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String = fieldName
-}
-
-class WordDelimiterConfigFieldMapping[T](
-    typeFieldDelimiter: WordDelimiter,
-    configFieldDelimiter: WordDelimiter) extends ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String =
-    (typeFieldDelimiter.toTokens _ andThen configFieldDelimiter.fromTokens _)(fieldName)
+trait ConfigFieldMapping[T] extends (String => String) {
+  def apply(fieldName: String): String
 }
 
 object ConfigFieldMapping extends LowPriorityConfigFieldMappingImplicits {
   def apply[T](implicit mapping: ConfigFieldMapping[T]): ConfigFieldMapping[T] = mapping
```

Comment:
It is not necessary, and I've removed it in 590e8bb. This is just a different implementation of `implicitly`. It allows you to get the implicit `ConfigFieldMapping` in scope by just writing, for example (in this case for `Int`), `ConfigFieldMapping[Int]`. This is more useful for `ConfigConvert` than it is for `ConfigFieldMapping`, so I've removed it.